internal/operator-controller/applier/helm.go

```diff
@@ -209,7 +213,119 @@ func (h *Helm) buildHelmChart(bundleFS fs.FS, ext *ocv1.ClusterExtension) (*char
 	if err != nil {
 		return nil, err
 	}
-	return h.BundleToHelmChartConverter.ToHelmChart(source.FromFS(bundleFS), ext.Spec.Namespace, watchNamespace)
+	var result *chart.Chart
+	contentType := bundleContentType(bundleFS)
+	switch contentType {
+	case HelmContent:
+		if !features.OperatorControllerFeatureGate.Enabled(features.HelmChartSupport) {
+			return nil, fmt.Errorf("helm chart support is not enabled")
+		}
+		result, err = loadChartWithEnrichments(bundleFS, WithInstallNamespace(ext.Spec.Namespace))
+		if err != nil {
+			return nil, fmt.Errorf("loading Helm chart; %w", err)
+		}
+	case BundleContent:
+		result, err = h.BundleToHelmChartConverter.ToHelmChart(source.FromFS(bundleFS), ext.Spec.Namespace, watchNamespace)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("unknown content found")
+	}
+
+	return result, nil
+}
+
+const (
+	HelmContent    string = "helm"
+	BundleContent  string = "bundle"
+	UnknownContent string = "unknown"
+)
+
+func bundleContentType(bundleFS fs.FS) string {
+	var contentType string
+	_ = fs.WalkDir(bundleFS, ".", func(path string, f fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !f.IsDir() {
+			// Check if Helm chart
+			if filepath.Dir(path) == "charts" &&
+				filepath.Ext(f.Name()) == ".tgz" {
+				contentType = HelmContent
+				return fs.SkipAll
+			}
+
+			// Check if registry/v1 bundle
+			if filepath.Dir(path) == "metadata" &&
+				f.Name() == "annotations.yaml" {
+				contentType = BundleContent
+				return fs.SkipAll
+			}
+		}
+
+		return nil
+	})
+
+	if contentType != "" {
+		return contentType
+	}
+
+	return UnknownContent
+}
+
+type ChartOption func(*chart.Chart)
+
+func WithInstallNamespace(namespace string) ChartOption {
+	re := regexp.MustCompile(`{{\W+\.Release\.Namespace\W+}}`)
+
+	return func(chrt *chart.Chart) {
+		for i, template := range chrt.Templates {
+			chrt.Templates[i].Data = re.ReplaceAll(template.Data, []byte(namespace))
+		}
+	}
+}
+
+func loadChartWithEnrichments(bundleFS fs.FS, options ...ChartOption) (*chart.Chart, error) {
+	chrt, err := loadPulledHelmChart(bundleFS)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, f := range options {
+		f(chrt)
+	}
+
+	return chrt, nil
+}
+
+func loadPulledHelmChart(bundleFS fs.FS) (*chart.Chart, error) {
```

Comment:
Why do we need to do this? 
What was your motivation?

Reply:
I think I got, I saw above loadChartWithEnrichments you can ignore this one thx.
Closing to facilitate the review for others.